### PR TITLE
Support `ignoredRequires` on `inline-requires` Plugin

### PR DESCRIPTION
**Pull request by yungsters**

### CODE REVIEW

File: packages/babel-preset-fbjs/plugins/__tests__/inline-requires-test.js [hunk 1]
```diff
@@ -1,180 +1,79 @@
...
+      snapshot: true,
+    },
+
+    'inlines `Foo` before it is declared': {
+      code: "const doSomething = () => Foo; const Foo = require('Foo');",
+      snapshot: true,
+    },
+
+    'inlines `Foo` when required via `customRequire`': {
+      code: "const Baz = require('Bar').Baz; Baz;",
```
Comment (rafeca) [inline packages/babel-preset-fbjs/plugins/__tests__/inline-requires-test.js]: To make this test match the description shouldn't the code be `const Baz = customRequire('Bar').Baz; Baz;`?

File: packages/babel-preset-fbjs/plugins/__tests__/__snapshots__/inline-requires-test.js.snap [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+"
+`;
+
+exports[`inline-requires inlines \`Foo\` properties with destructuring: inlines \`Foo\` properties with destructuring 1`] = `
+"
+const {Baz} = require('Foo'); Baz;
+
+      ↓ ↓ ↓ ↓ ↓ ↓
+
+const { Baz } = require('Foo');Baz;
```
Comment (rafeca) [inline packages/babel-preset-fbjs/plugins/__tests__/__snapshots__/inline-requires-test.js.snap]: This require should be inlined... 😕

Comment (mjesun) [inline packages/babel-preset-fbjs/plugins/__tests__/__snapshots__/inline-requires-test.js.snap]: This is incorrect.

**Comment by rafeca** [comment]
Based on the tests, it seems that the plugin does not inline destructured requires... that's weird...

Can you add a couple more tests?

- One that checks that removes a `require` if it's not used
- Another that checks that leaves a `require` if it's unassigned

File: packages/babel-preset-fbjs/plugins/__tests__/__snapshots__/inline-requires-test.js.snap [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+"
+`;
+
+exports[`inline-requires inlines \`Foo\` when required via \`customRequire\`: inlines \`Foo\` when required via \`customRequire\` 1`] = `
+"
+const Baz = require('Bar').Baz; Baz;
+
+      ↓ ↓ ↓ ↓ ↓ ↓
+
+const Baz = require('Bar').Baz;Baz;
```
Comment (mjesun) [inline packages/babel-preset-fbjs/plugins/__tests__/__snapshots__/inline-requires-test.js.snap]: This is incorrect too.

**Comment by mjesun** [comment]
It looks like most of the cases the old tests covered were removed (e.g. integration with import commonjs interoperability, and property access). Could you add all of the existing back?

File: packages/babel-preset-fbjs/plugins/__tests__/__snapshots__/inline-requires-test.js.snap (hunk 2 above)
Comment (mjesun) [reply packages/babel-preset-fbjs/plugins/__tests__/__snapshots__/inline-requires-test.js.snap]: Note: that's per se _not_ incorrect, as we rely on the destructuring Babel transform to lower `const {a: c} = require('b')` to `const c = require('b').a`; but that's however not what the existing test asserted.

**Comment by yungsters** [comment]
I hope the new version of this pull request is easier to follow. It only introduces two new tests, and all existing tests continue to pass.

**Pull request merged**